src/discussions/common/Confirmation.jsx

```diff
@@ -26,25 +29,27 @@ function DeleteConfirmation({
       </ModalDialog.Body>
       <ModalDialog.Footer>
         <ActionRow>
-          <ModalDialog.CloseButton variant="tertiary">
+          <ModalDialog.CloseButton variant={!ifDeleteConfirmation ? 'tertiary' : 'default'}>
             {intl.formatMessage(messages.deleteConfirmationCancel)}
           </ModalDialog.CloseButton>
-          <Button variant="primary" onClick={onDelete}>
-            {intl.formatMessage(messages.deleteConfirmationDelete)}
+          <Button variant={!ifDeleteConfirmation ? 'primary' : 'danger'} onClick={comfirmAction}>
+            {!ifDeleteConfirmation
+              ? intl.formatMessage(messages.deleteConfirmationDelete)
+              : intl.formatMessage(messages.reportConfirmationConfirm) }
```

Comment:
Make this component more generic. Pass button variants or messages as props instead of adding conditions.

Reply:
and make these props to optional.